container_pipeline/lib/openshift.py

```diff
@@ -0,0 +1,77 @@
+import logging
+import time
+
+from container_pipeline.lib import settings
+
+from .lib.command import run
+
+
+class OpenshiftError(Exception):
+    pass
+
+
+class Openshift(object):
+
+    def __init__(self, endpoint=None, config=None, cert=None, logger=None):
+        self.endpoint = endpoint or settings.OPENSHIFT_ENDPOINT
+        self.config = config or settings.OC_CONFIG
+        self.cert = cert or settings.OC_CERT
+        self.oc_cmd_suffix = (
+            '--config {config} --certificate-authority {cert}'.format(
+                config=self.config, cert=self.cert))
+        self.logger = logger or logging.getLogger('oc')
+
+    def login(self, user=None, password=None):
+        user = user or settings.OPENSHIFT_USER
+        password = password or settings.OPENSHIFT_PASSWORD
+        self.logger.debug(
+            run('oc login {endpoint} -u {user} -p {password} {suffix}'.format(
+                    endpoint=self.endpoint, user=self.user,
+                    password=self.password, suffix=self.oc_cmd_suffix
+                ))
+        )
+
+    def use_project(self, project):
+        self.logger.debug(
+            'oc project {project} {suffix}'.format(
+                project=project, suffix=self.oc_cmd_suffix))
+
+    def build(self, project, build):
+        output = run(
+            'oc --namespace {project} start-build {build} '
+            '{suffix}'.format(
+                project=project, build=build, suffix=self.oc_cmd_suffix)
+        )
+        build_id = output.split('"')[-1].rstrip()
+        return build_id
+
+    def get_build_status(self, project, build_id):
+        output = run(
+            'oc get --namespace {project} build/{build_id} {suffix} | '
+            'grep -v STATUS'.format(
+                project=project, build_id=build_id)
+        )
+        return output.split()[3]
+
+    def wait_for_build_status(self, project, build_id, status, retries=None,
```

Comment:
We are not providing the value for retires from build? and also there is no certain number, it will take to get the build complete.